### ISSUE TITLE
Add table driven execution at scenario level

### ISSUE THREAD

**Issue by kashishm** [getgauge/gauge#111]

**Comment by dhdeshpa** [comment]
When can we expect this feature to be released? We want to use data driven tests for one scenario for which this is required.

**Comment by MaxBt** [comment]
Is there any update?

This issue is one of main drawbacks to move tests from Cucumber to Gauge.

Would be great to have table for scenario even without table driven execution at scenario level, i.e. If scenario is executed, it is executed for all table.
{code}
Java arithmetic check
----------------

    | number | square |
    | 2 | 4 |
    | 3 | 9 |
    | 4 | 16 |

* raise <number> to 2 power 
* do nothing with  <square>
* check that <number> in 2 degree is equal to <square>
{code}

**Comment by sriv** [comment]
@MaxBt - there has been a similar discussion on #604. This is something we feel is worth taking on, but I can't give an ETA yet. 

Please feel free to add your thoughts on #604 or here.

**Comment by florinalex1984** [comment]
Hi,
Any update for this issue?

**Comment by zabil** [comment]
Will pick this up first thing after 1.0 release.

**Comment by nehashri** [comment]
This feature is toggled off by default. To use this feature set the `allow_scenario_datatable` environment variable to `true`.

**Comment by sguptatw** [comment]
This is more of an experimental feature. Hence toggled off by default.
- The execution statistics do not show the details of the table scenarios explicitly. Any failure in one of the cases is shown as a scenario failure
- If there is column in the data table at spec level with the same name as the scenario data table column name. The scenario table values are considered.
- The value in the spec level data table does not get resolved if used inside a scenario data table.
- Even if scenario table value is not used inside a scenario, a given scenario executes spec-level-table * scenario-level-table number of times.
- File parameter inside the scenario level table does not get resolved even though currently it does not show any diagnostics. The file parameter inside the spec level table gets resolved as earlier. 
> Note: Scenario table values cannot be used in context or teardown steps. However the spec level table values can continue to be used.

**Comment by florinalex1984** [comment]
Hi,

Having the following scenario,
## Vowel counts in multiple word test test
* Almost all words have vowels
     |Word  |Vowel Count|
     |------|-----------|
     |ABC |3          |
     |CDE|2           |
     |EFG  |1         |
* Test data

- how can i make this scenario to run for every line? (every table row = 1 test scenario?)

**Comment by florinalex1984** [comment]
Yes, but what if i want to use the same step for tables that have different number of columns?
With this implementation, i have to write steps for tables with 2 columns, steps for tables with 3 columns and so forth. 
Is it possible to refer to the entire table header and the row which is tested?

**Comment by sriv** [comment]
Table row reference isn't supported in Gauge, and it's not just this feature. The best option is to pass a table, but that cannot be clubbed with data driven execution at the moment.

I.e. steps need to explicitly use column headers, and it is not mandatory for steps to use all column headers, however a step cannot have a header that is not defined in scenario/spec.

Please feel free to log a separate issue and we can discuss this at length.

**Comment by dchoub** [comment]
> Hi @florinalex1984
> 
> This scenario by default runs only once:
> 
> ```
> ## Vowel counts in multiple word
> 
> This is the second scenario in this specification
> 
> Here's a step that takes a table
> 
> * Almost all words have vowels
>      |Word  |Vowel Count|
>      |------|-----------|
>      |Gauge |3          |
>      |Mingle|2          |
>      |Snap  |1          |
>      |GoCD  |1          |
>      |Rhythm|0          |
> ```
> 
> Changing it to this:
> 
> ```
> ## Vowel counts in multiple word
> 
>      |Word  |Vowel Count|
>      |------|-----------|
>      |Gauge |3          |
>      |Mingle|2          |
>      |Snap  |1          |
>      |GoCD  |1          |
>      |Rhythm|0          |
> 
> This is the second scenario in this specification
> 
> Here's a step that takes a table
> 
> * The word <Word> has <Vowel Count> vowels.
> ```
> 
> and running it by setting `allow_scenario_datatable` env variable should run this scenario multiple times.
> 
> ex:
> 
> `allow_scenario_datatable=true gauge run specs` yields this output for me:
> 
> ```
> # Specification Heading
>   ## Vowel counts in single word	 ✔ ✔
> 
>    |Word |Vowel Count|
>    |-----|-----------|
>    |Gauge|3          |
>   ## Vowel counts in multiple word	 ✔ ✔
> 
>    |Word  |Vowel Count|
>    |------|-----------|
>    |Mingle|2          |
>   ## Vowel counts in multiple word	 ✔ ✔
> 
>    |Word|Vowel Count|
>    |----|-----------|
>    |Snap|1          |
>   ## Vowel counts in multiple word	 ✔ ✔
> 
>    |Word|Vowel Count|
>    |----|-----------|
>    |GoCD|1          |
>   ## Vowel counts in multiple word	 ✔ ✔
> 
>    |Word  |Vowel Count|
>    |------|-----------|
>    |Rhythm|0          |
>   ## Vowel counts in multiple word	 ✔ ✔
> 
> Successfully generated html-report to => /tmp/foo/reports/html-report/index.html
> Specifications:	1 executed	1 passed	0 failed	0 skipped
> Scenarios:	2 executed	2 passed	0 failed	0 skipped
> ```

Thanks @sriv  it worked for me but in report its giving total scenario count correct but in Passed column count is only one. its not considering the each row of table as significant scenario
